src/Filesystem/JsonFileReader.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Kollex\Filesystem;
+
+use Assert\Assertion;
+use Kollex\Assortment\BaseProductPackagingType;
+use Kollex\Assortment\BaseProductUnitType;
+use Kollex\Assortment\PackagingType;
+use Kollex\Assortment\Product;
+use Kollex\Assortment\ProductInterface;
+
+class JsonFileReader implements FileReaderInterface
+{
+    /**
+     * @return ProductInterface[]
+     */
+    public function read(string $filename): iterable
+    {
+        $data = file_get_contents($filename);
+        $json = json_decode($data, true, 4, JSON_THROW_ON_ERROR);
+
+        Assertion::keyExists($json, 'data');
```

Comment:
Validation is better to move in another layer and validate not just `data` key, because we will get warnings every time when a key doesn't exist.